reaper/reaper.go

```diff
@@ -79,45 +79,59 @@ func (reaper reaper) getPods() *v1.PodList {
 	return podList
 }
 
-func (reaper reaper) reapPod(pod v1.Pod, reasons []string) {
+func (reaper reaper) reapPod(pod v1.Pod, reasons []string, reapedPods int) {
 	deleteOptions := &metav1.DeleteOptions{
 		GracePeriodSeconds: reaper.options.gracePeriod,
 	}
+
+	podLog := logrus.WithFields(logrus.Fields{
+		"pod":     pod.Name,
+		"reasons": reasons,
+	})
+
 	if reaper.options.dryRun {
-		logrus.WithFields(logrus.Fields{
-			"pod":     pod.Name,
-			"reasons": reasons,
-		}).Info("pod would be reaped but pod-reaper is in dry-run mode")
+		podLog.Info("pod would be reaped but pod-reaper is in dry-run mode")
+
+		return
+	}
+
+	if reaper.options.maxPods > 0 && reapedPods >= reaper.options.maxPods {
+		podLog.WithFields(logrus.Fields{
+			"reapedPods": reapedPods,
+			"maxPods":    reaper.options.maxPods,
+		}).Info("pod would be reaped but maxPods is exceeded")
+
+		return
+	}
+
+	podLog.Info("reaping pod")
```

Comment:
So only thing I see, and it's just a nitpick, is related to consistency of structured log messages. I'm leaning towards favoring detailed, structured wherever logging is happening.